docs/proposals/2019-07-12_Image_Focal_Point_Proposal.md

```diff
@@ -0,0 +1,255 @@
+# FOCAL POINT PROPOSAL
+
+## Problem
+Editorial users wish to be able to declare a focal point for images. Additionally, these users wish to be able to set a focal point over-ride for images when used in a Gallery, Story, Video, etc.
+
+## Usage Patterns
+- Anglerfish users will set a global focal point for an image
+- Anglerfish users can set a focal point over-ride for a gallery cover image
+- Ellipsis users can set a focal point on any image, be it featured image, or an image in the body
+- Goldfish users can set a focal point on the video cover image
+
+## Proposal
+We propose adding a focal point trait for use in images. The focal point is a single pixel position identified on an image will be stored by a set of two integer values: X and Y coordinates. The focal point field will appear in the top-level of image ANS.
+
+Parent documents can override an image's focal point. Gallery overrides will appear in the image's additional properties (as caption/headline overrides are). Story and thumbnail overrides will be done through referent_properties on the image.
```

Comment:
Any reasons why we don't wanna use the same way (referent_properties) for Gallery?

Reply:
I have proposed this and @gengel made an argument that it will be difficult to implement.

Reply:
I'd also be curious why ... it seems like with this implementation, downstream consumers (e.g. front-end websites) will have to know that they need to treat focal point data differently when it comes from a gallery vs. when it comes from anything else.

Reply:
@chenghsinw In an ideal world, yes, galleries would store `reference`s to images and put their overrides in `referent_properties`

However, for better or for worse, Anglerfish galleries use `image` objects with a subset of `image` properties in them. These image objects are stored as-is in Content API and not denormalized (since, in a way, they already are.) 

For downstream consumers inspecting a story or a video, focal point overrides are completely transparent. The consumer simply checks `.focal_point` and the correct value is in place based on the context.  However, with galleries, the consumer still has to check in two places for the focal point value -- `focal_point` and `referent_properties.focal_point`.  So changing the field name, IMHO, doesn't bring more clarity or consistency. On top of that, the Anglerfish paradigm to-date for field overrides has been to put them in fieldnameOverride. Changing that means there's now an inconsistency between old fields in AF and new fields in AF, on top of the inconsistency between AF and other apps.

@chenghsinw If you see a way around this, I am open to suggestions

Reply:
@gengel Since u mentioned the `fieldnameOverride`, how does it work today? I don’t see it’s in the ans schema.

Reply:
nvm, I think u're referring the overriding using the additional_properties

Reply:
I can see the challenge here, but personally I think this kind of rules should be explicitly defined in the schema, so this not the knowledge only a small portion of ppl know. Base on my experience with the PS side, when they work on data ingestion, this ANS schema is the info they rely on heavily. The different behavior for handling field overriding can be overlooked easily.

Reply:
100% agree. I didn't even realize this was the behavior until I started digging into this proposal